### PR TITLE
build: `flutter pub upgrade` and `pod update`

### PR DESCRIPTION
**Pull request by vsomayaji**

### CODE REVIEW

File: example/android/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.0'
+        classpath 'com.android.tools.build:gradle:7.1.2'
```
Comment (vsomayaji) [inline example/android/build.gradle:9]: From https://medium.com/flutter/whats-new-in-flutter-3-8c74a5bc32d0:

> If you create a new project with the Flutter tool, you might notice that the generated files now use the latest versions of the Gradle and Android Gradle plugins. For existing projects, you need to manually bump the versions to 7.4 for Gradle, and 7.1.2 for the Android Gradle plugin.

File: example/ios/Runner/Info.plist [hunk 2]
```diff
@@ -43,5 +43,7 @@
 	</array>
 	<key>UIViewControllerBasedStatusBarAppearance</key>
 	<false/>
+	<key>CADisableMinimumFrameDurationOnPhone</key>
```
Comment (vsomayaji) [inline example/ios/Runner/Info.plist:46]: This was added after running `flutter build ipa`.

File: example/lib/main.dart [hunk 3]
```diff
@@ -463,7 +463,7 @@ class _MyAppState extends State<MyApp> implements RoomHandler {
     _platformClient = null;
 
     // Close the call progress dialog if it's open.
-    if (!ModalRoute.of(context)!.isCurrent) {
+    if (mounted && !ModalRoute.of(context)!.isCurrent) {
```
Comment (vsomayaji) [inline example/lib/main.dart:466]:
```
   info • Do not use BuildContexts across async gaps • lib/main.dart:466:10 • use_build_context_synchronously
   info • Do not use BuildContexts across async gaps • lib/main.dart:467:7 • use_build_context_synchronously
```

File: example/web/index.html [hunk 4]
```diff
@@ -31,74 +31,28 @@
 
   <title>example</title>
   <link rel="manifest" href="manifest.json">
+
+  <script>
```
Comment (vsomayaji) [inline example/web/index.html:35]: This is the new index.html page for Flutter 3.

https://docs.flutter.dev/development/platform-integration/web/initialization#getting-started

**Review by ipainchaud — APPROVED** [review]

**Pull request merged**